sklearnex/cluster/k_means.py

```diff
@@ -41,7 +41,12 @@
     from .._device_offload import dispatch, wrap_output_data
     from .._utils import PatchingConditionsChain
 
-    @control_n_jobs(decorated_methods=["fit", "predict", "transform", "fit_transform"])
+    if sklearn_check_version("1.6"):
+        from sklearn.utils.validation import validate_data
+    elif sklearn_check_version("1.0"):
+        validate_data = sklearn_KMeans._validate_data
+
```

Comment:
Closed `if-else` without branch with undefined validate_data is preferred since sklearn<1.0 support is dropped.

Reply:
@Alexsandruss done!